packages/network/src/types.ts

```diff
@@ -113,12 +113,35 @@ export type SystemCall<C extends Components = Components> = {
   updates: NetworkComponentUpdate<C>[];
 };
 
+export type NetworkEphemeralComponentUpdate<C extends Components = Components> = {
+  [key in keyof C]: {
+    type: NetworkEvents.NetworkEphemeralComponentUpdate;
+    component: key & string;
+    value: ComponentValue<SchemaOf<C[key]>> | undefined;
+    partialValue?: Partial<ComponentValue<SchemaOf<C[key]>>>;
+    initialValue?: ComponentValue<SchemaOf<C[key]>>;
+  };
+}[keyof C] & {
+  entity: EntityID;
+  lastEventInTx: boolean;
+  txHash: string;
+  txMetadata?: TxMetadata;
+  blockNumber: number;
+  // TODO: make this required, so we can later sort by logIndex when concatenating event types
+  //       would require updating the ECS snapshot, though
+  logIndex?: number;
+};
```

Comment:
do we need a separate event for this? why can't we lean on `NetworkComponentUpdate` as it is? maybe so we can discern whether or not this record should be cached downstream?

Reply:
yes its purely for caching. i agree it's a bit dumb.

Reply:
I added `partialValue` and `initialValue` to `NetworkComponentUpdate` for the purposes of v2. Another route is to extend `NetworkComponentUpdate` with something like `cache?: boolean` where it defaults to true and you can explicitly set to `false` for ephemeral events.

Reply:
would it work if we add `ephemeral: boolean` to `NetworkComponentUpdate` instead of a new event? Seems like that would simplify a lot of the logic/types below

(would tend to `ephemeral` instead of `cache` bc it would be used both for caching inside the sync logic, but also for downstream consumers like in `setupMUDV2Network`/ `applyNetworkUpdates`)

Reply:
ok wil add ephemeral 👍🏼